pkg/utils/kubernetes/client/client.go

```diff
@@ -265,6 +265,47 @@ func DefaultGoneEnsurer() GoneEnsurer {
 	return defaultGoneEnsurer
 }
 
+// GoneBeforeEnsurer returns an implementation of `GoneEnsurer` which is time aware.
+func GoneBeforeEnsurer(before time.Time) GoneEnsurer {
+	return &beforeGoneEnsurer{
+		time: before,
+	}
+}
+
+type beforeGoneEnsurer struct {
+	time time.Time
+}
+
+// EnsureGoneBefore ensures that no given object or objects in the list exist before the given time.
+func (b *beforeGoneEnsurer) EnsureGone(ctx context.Context, c client.Client, obj runtime.Object, opts ...client.ListOption) error {
+	if err := EnsureGone(ctx, c, obj, opts...); err != nil {
+		if _, ok := err.(*objectsRemaining); !ok {
+			return err
+		}
+		var (
+			relevants  []runtime.Object
+			remainings = err.(*objectsRemaining)
+		)
+
+		for _, remaining := range *remainings {
+			accessor, err := meta.Accessor(remaining)
+			if err != nil {
+				return err
+			}
+
+			if accessor.GetCreationTimestamp().Time.Before(b.time) {
```

Comment:
To be safe, convert everything to UTC

Reply:
AFAIK, the `creationTimestamp` as well as `deletionTimestamp` don't contain any time zone information.

Reply:
Yep, according to the documentation the `creationTimestamps` is saved in UTC. I guess the same goes for the `deletionTimestamp`.